inferentia/scripts/gen_triton_model.py

```diff
@@ -261,29 +220,28 @@ def execute(self, requests):
         responses = []
 
         for request in requests:
-            num_threads = self.num_threads
             inputs = []
-            for name in self.input_dict.keys():
+            for i in range(len(self.input_dict)):
+                name, dt, shape = self.input_dict[i]
                 tensor = pb_utils.get_input_tensor_by_name(request,
                                                            name).as_numpy()
-                inputs.append(torch.LongTensor(tensor))
-            results = self.model_neuron(inputs)
+                inputs.append(torch.as_tensor(tensor))
+
+            results = self.model_neuron(*inputs)
 
             output_tensors = []
-            for name in self.output_dict.keys():
-                result_shards = []
-                for i in range(num_threads):
-                    result_shards.append(results[i][len(output_tensors)])
-                merged_result = np.concatenate(result_shards, axis=0)
-                dt, shape = self.output_dict[name]
-                output_tensor = pb_utils.Tensor(name,
-                                           merged_result.astype(pb_utils.triton_string_to_numpy(dt)))
+            for i in range(len(self.output_dict)):
```

Comment:
There is an edge case where the outputs in model config can be ouput__0 and output__2 (1 is skipped). In this case range would not be a good idea.

Reply:
Why not just use for i in self.output_dict? Similarly for input.
We do not care the order in which the outputs are read as long as they use the right index.

Reply:
We weed out the edge case in the validation logic in __validate_input_dict /validate_output_dict..
Basically, we don't support skipping of indices.. 

>We do not care the order in which the outputs are read as long as they use the right index.
I think that is true.. I will fix it..

Reply:
> Basically, we don't support skipping of indices..

Why do we want to skip it? The user should be allowed to ask for a subset of the outputs from the model, like with all other backends

Reply:
We can not do this for inputs as the order *does* matter for placing the input in the list.